src/services/address.service.ts

```diff
@@ -38,35 +38,98 @@ class AddressService {
     }
 
     public async distance(addressRequest?: any): Promise<any> {
-        // Complete this
+        return new Promise<any>(async (resolve, reject) => {
+            this.request(addressRequest)
+            .then((response) => {
+                if(response.length < 2) {
+                    loggerService.warning({ path: "/address/distance", message: "Search conditions did not match to at least 2 addresses." }).flush();
+                    reject("Search conditions must match to at least 2 addresses");
+                }
+
+                const address1 = response[0];
+                const address2 = response[1];
+
+                var lat1 = address1["latitude"];
+                var long1 = address1["longitude"];
+                var lat2 = address2["latitude"];
+                var long2 = address2["longitude"];
+
+                var dist = this.getDistance(lat1, long1, lat2, long2);
+                var distKM = dist;
+                var distMI = dist * 0.621;
+
+                resolve({
+                    "kilometers": distKM,
+                    "miles": distMI
+                });
```

Comment:
What if the requester wants one or the other? Could we parse that from the request?